pkg/dns/daemon.go

```diff
@@ -61,71 +62,73 @@ func (d *Daemon) internal(ctx context.Context) (bool, error) {
 		return true, nil
 	}
 
-	logrus.Infof("Renewing and syncing AcornDNS")
+	logrus.Infof("Renewing and syncing AcornDNS...")
 
-	dnsSecret := &corev1.Secret{}
-	err = d.client.Get(ctx, router.Key(system.Namespace, system.DNSSecretName), dnsSecret)
+	dnsSecret, err := d.getDNSSecret(ctx)
 	if err != nil {
-		if apierrors.IsNotFound(err) {
-			logrus.Infof("DNS secret %v/%v not found, not proceeding with DNS renewal", system.Namespace, system.DNSSecretName)
-			return true, nil
-		}
-		logrus.Errorf("Problem getting DNS secret %v/%v: %v", system.Namespace, system.DNSSecretName, err)
+		logrus.Errorf("Failed to get DNS secret: %v", err)
 		return false, nil
 	}
 
-	var domain, token string
-	domain = string(dnsSecret.Data["domain"])
-	token = string(dnsSecret.Data["token"])
+	domain, token := string(dnsSecret.Data["domain"]), string(dnsSecret.Data["token"])
 	if domain == "" || token == "" {
-		logrus.Errorf("DNS secret %v/%v exists but is missing domain (%v) or token", system.Namespace, system.DNSSecretName, domain)
-		return false, nil
+		return false, fmt.Errorf("DNS secret %v/%v exists but is missing domain (%v) or token", system.Namespace, system.DNSSecretName, domain)
 	}
 
-	var ingresses netv1.IngressList
-	err = d.client.List(ctx, &ingresses, &kclient.ListOptions{
-		LabelSelector: klabels.SelectorFromSet(map[string]string{
-			labels.AcornManaged: "true",
-		}),
-	})
-	if err != nil {
-		logrus.Errorf("Failed to list ingresses for DNS renewal: %v", err)
+	if err := d.syncIngress(ctx, domain, token, *cfg.AcornDNSEndpoint, dnsSecret); err != nil {
+		logrus.Errorf("Failed to sync ingress: %v", err)
 		return false, nil
 	}
 
-	var recordRequests []RecordRequest
-	ingressMap := make(map[FQDNTypePair]netv1.Ingress)
-	for _, ingress := range ingresses.Items {
-		rrs, _ := ToRecordRequestsAndHash(domain, &ingress)
-		recordRequests = append(recordRequests, rrs...)
-		for _, r := range rrs {
-			ingressMap[FQDNTypePair{FQDN: r.Name, Type: string(r.Type)}] = ingress
+	logrus.Infof("Renewed and synced AcornDNS!")
+
+	return true, nil
+}
+
+func (d *Daemon) getDNSSecret(ctx context.Context) (*corev1.Secret, error) {
+	dnsSecret := &corev1.Secret{}
+	err := d.client.Get(ctx, router.Key(system.Namespace, system.DNSSecretName), dnsSecret)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return dnsSecret, fmt.Errorf("DNS secret %v/%v not found, not proceeding with DNS renewal", system.Namespace, system.DNSSecretName)
 		}
+		return nil, fmt.Errorf("problem getting DNS secret %v/%v: %v", system.Namespace, system.DNSSecretName, err)
+	}
+	return dnsSecret, nil
+}
+
+func (d *Daemon) syncIngress(ctx context.Context, domain, token, acornDNSEndpoint string, secret *corev1.Secret) error {
+	var ingress netv1.Ingress
+	err := d.client.Get(ctx, router.Key(system.Namespace, system.DNSIngressName), &ingress)
+	if err != nil {
```

Comment:
You should probably check for a notFound error and treat it different by not returning an error.

You can validly get a NotFound if you are running locally with cfg.AcornDNS == auto. This should result in no ingress being created.

Reply:
Gotcha. This has been updated.